django_jalali/db/models.py

```diff
@@ -34,6 +34,18 @@ def filter(self, *args, **kwargs):
                     last_day = 30
                 new_kwargs['%s__lte' % filed_name[0]] = jdatetime.datetime(
                     int(kwargs[k]), 12, last_day, 23, 59, 59)
+            elif "__date" in k:
+                jdate = kwargs[k]
+                if isinstance(jdate, jdatetime.datetime):
+                    gregorian_date = jdate.date().togregorian()
+                elif isinstance(jdate, jdatetime.date):
+                    gregorian_date = jdate.togregorian()
+                elif isinstance(jdate, str):
+                    year, month, day = map(int, jdate.split('-'))
+                    gregorian_date = jdatetime.date(year, month, day).togregorian()
+                else:
+                    raise TypeError(f"Value of `__date` filter must be an instance of jdatetime.datetime, jdatetime.date or str")
```

Comment:
The `f` is not needed here:
```suggestion
                    raise TypeError("Value of `__date` filter must be an instance of jdatetime.datetime, jdatetime.date or str")
```

Reply:
Yeah I forgot to remove that, I want to do `Expected jdatetime.datetime, jdatetime.date or str, got {}.`.
What do you think?

Reply:
ok let's do it like
```
raise TypeError(f"`__date` filter Expected jdatetime.datetime, jdatetime.date or str, got `{jdate.__class__.__name__}`.")
```